src/main/java/edu/unc/lib/boxc/migration/cdm/CdmExportCommand.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright 2008 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.boxc.migration.cdm;
+
+import static edu.unc.lib.boxc.migration.cdm.util.CLIConstants.outputLogger;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.concurrent.Callable;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpHost;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+import edu.unc.lib.boxc.migration.cdm.exceptions.MigrationException;
+import edu.unc.lib.boxc.migration.cdm.model.MigrationProject;
+import edu.unc.lib.boxc.migration.cdm.services.CdmExportService;
+import edu.unc.lib.boxc.migration.cdm.services.CdmFieldService;
+import edu.unc.lib.boxc.migration.cdm.services.MigrationProjectFactory;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.ParentCommand;
+
+/**
+ * @author bbpennel
+ */
+@Command(name = "export",
+        description = "Export records for a collection from CDM")
+public class CdmExportCommand implements Callable<Integer> {
+    @ParentCommand
+    private CLIMain parentCommand;
+
+    @Option(names = { "--cdm-url" },
+            description = {"Base URL to the CDM web service API. Falls back to CDM_BASE_URL env variable.",
+                    "Default: ${DEFAULT-VALUE}"},
+            defaultValue = "${env:CDM_BASE_URL:-http://localhost:82/}")
+    private String cdmBaseUri;
+    @Option(names = { "-u", "--cdm-user"},
+            description = {"User name for CDM requests.",
+                    "Defaults to current user: ${DEFAULT-VALUE}"},
+            defaultValue = "${sys:user.name}")
+    private String cdmUsername;
+    @Option(names = {"-p", "--cdm-password"},
+            description = "Password for CDM requests. Required.",
+            arity = "0..1",
```

Comment:
So, this can take 0 or 1 argument?

Reply:
Sort of, this lets you either provide your password directly as the value of the option, or if you don't include a value it will prompt you to type in your password.